core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -52,6 +54,7 @@
 import org.apache.hadoop.fs.FSDataInputStream;
 import org.apache.hadoop.fs.FSDataOutputStream;
 import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.LocatedFileStatus;
```

Comment:
which hadoop version introduces `org.apache.hadoop.fs.LocatedFileStatus`?
If it is a recent version, by having this line of src code, we might not be able to compile Alluxio client with earlier hadoop version. In this case, we may need to introduce multiple implementations of `AbstractFileSystem.java` for different Hadoop versions.

Reply:
`hadoop 2.0 +` version introduces `org.apache.hadoop.fs.LocatedFileStatus`?

Reply:
It looks like this is introduced in 2.x. So without your change @witgo , does this prevent you using any data source for Spark? If that is the case, please let us know and we will see the importance and address it accordingly.